pkg/services/dashboards/database/database_test.go

```diff
@@ -639,6 +639,41 @@ func TestIntegrationDashboard_Filter(t *testing.T) {
 	assert.Equal(t, dashB.ID, results[0].ID)
 }
 
+func TestGetExistingDashboardByTitleAndFolder(t *testing.T) {
+	sqlStore := db.InitTestDB(t)
+	cfg := setting.NewCfg()
+	cfg.IsFeatureToggleEnabled = func(key string) bool { return false }
+	quotaService := quotatest.New(false, nil)
+	dashboardStore, err := ProvideDashboardStore(sqlStore, cfg, testFeatureToggles, tagimpl.ProvideService(sqlStore, cfg), quotaService)
+	require.NoError(t, err)
+	insertTestDashboard(t, dashboardStore, "Apple", 1, 0, false)
+	t.Run("Should be able to get dashboard with existing name", func(t *testing.T) {
+		err = sqlStore.WithDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+			_, err = getExistingDashboardByTitleAndFolder(sess, &dashboards.Dashboard{Title: "Apple", OrgID: 1}, sqlStore.GetDialect(), false, false)
+			return err
+		})
+		require.Error(t, err)
+	})
+
+	t.Run("Should not be able to get dashboard with non-existing name", func(t *testing.T) {
+		err = sqlStore.WithDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+			_, err = getExistingDashboardByTitleAndFolder(sess, &dashboards.Dashboard{Title: "Beta"}, sqlStore.GetDialect(), false, false)
+			return err
+		})
+		require.NoError(t, err)
+	})
+
+	t.Run("Should be able to get dashboard with existing name in a folder", func(t *testing.T) {
+		savedFolder := insertTestDashboard(t, dashboardStore, "test dash folder", 1, 0, true, "prod", "webapp")
+		savedDash := insertTestDashboard(t, dashboardStore, "test dash", 1, savedFolder.ID, false, "prod", "webapp")
+		err = sqlStore.WithDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+			_, err = getExistingDashboardByTitleAndFolder(sess, savedDash, sqlStore.GetDialect(), false, false)
```

Comment:
```suggestion
			_, err = getExistingDashboardByTitleAndFolder(sess, &dashboards.Dashboard{Title: savedDash.Title, FolderID: savedFolder.ID, OrgID: 1}, sqlStore.GetDialect(), false, false)
```
same as the 1st test, we want to test [this](https://github.com/grafana/grafana/blob/b80c20f80ba0d0de223d81e912c9fe57224c72c9/pkg/services/dashboards/database/database.go#L417) specific case (successfully retrieving an existing dashboard under a folder by title and folder identifier):
so you can better pass a `*dashboards.Dashboard` where only the org, title and folder identifier are set: `savedDash` has other fields set (eg slug)
also, same as above better to check for the specific `dashboards.ErrDashboardWithSameNameInFolderExists` error.